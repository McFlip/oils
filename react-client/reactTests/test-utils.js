import { render } from 'react-testing-library';
import React from 'react';
import { createStore, applyMiddleware } from 'redux';
import promise from "redux-promise";
import reducers from 'reducers/';
import { Provider } from 'react-redux';
import { BrowserRouter } from "react-router-dom";

function renderWithRedux(
  ui,
  {initialState, reducer = (state = {}) => state, store = createStore(reducer, initialState)} = {}) {
  return {
    ...render(
      <Provider store={store}>
        <BrowserRouter>
          {ui}
        </BrowserRouter>
      </Provider>
    ),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store,
  }
}

export const newStore = () => {
  const createStoreWithMiddleware = applyMiddleware(promise)(createStore);
  const store = createStoreWithMiddleware(reducers, {});
  return store;
}


// re-export everything
export * from 'react-testing-library';

// override render method
export {renderWithRedux as render};
