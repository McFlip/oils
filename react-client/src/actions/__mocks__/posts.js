/* eslint-env jest */
import { FETCH_POST, CREATE_POST, DELETE_POST } from 'constants/'
import { testState } from './testData'

const { prods: { a: { posts } } } = testState

export const fetchPost = jest.fn(() => {
  return {
    type: FETCH_POST,
    payload: { data: posts[0] }
  }
})

export const createPost = jest.fn(() => {
  return {
    type: CREATE_POST,
    payload: 'post created'
  }
})

export const deletePost = jest.fn(() => {
  return {
    type: DELETE_POST,
    payload: 'post deleted'
  }
})

export const updatePost = jest.fn(() => {
  return {
    type: CREATE_POST,
    payload: 'post created'
  }
})
