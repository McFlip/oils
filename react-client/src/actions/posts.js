import axios from 'axios'

export const FETCH_POSTS = 'fetch_posts'
export const FETCH_POST = 'fetch_post'
export const CREATE_POST = 'create_post'
export const DELETE_POST = 'delete_post'

const ROOT_URL = 'http://localhost:3000'

export function fetchPosts () {
  const request = axios.get(`${ROOT_URL}/posts`)

  return {
    type: FETCH_POSTS,
    payload: request
  }
}

export function createPost (values, callback) {
  // console.log(values)
  const config = {     
    headers: { 'content-type': 'multipart/form-data' }
  }
  const request = axios
    .post(`${ROOT_URL}/posts`, values, config)
    .then(() => callback())

  return {
    type: CREATE_POST,
    payload: request
  }
}

export function updatePost (values, callback) {
  console.log(values)
  const config = {     
    headers: { 'content-type': 'multipart/form-data' }
  }
  const request = axios
    .post(`${ROOT_URL}/posts/${values.get('postId')}`, values, config)
    .then(() => callback())

  return {
    type: CREATE_POST,
    payload: request
  }
}

export function fetchPost (prodId, postId) {
  const request = axios.get(`${ROOT_URL}/products/${prodId}/posts/${postId}`)

  return {
    type: FETCH_POST,
    payload: request
  }
}

export function deletePost (id, callback) {
  const request = axios
    .delete(`${ROOT_URL}/posts/${id}`)
    .then(() => callback())

  return {
    type: DELETE_POST,
    payload: id
  }
}
