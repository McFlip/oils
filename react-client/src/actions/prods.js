import axios from 'axios'
import { FETCH_PRODS, FETCH_PROD, DELETE_PROD, ROOT_URL, JWT } from '../constants/'

const config = {
  headers: { 'Authorization': `Bearer ${JWT}` }
}

export function fetchProds () {
  const request = axios.get(`${ROOT_URL}/products`, config)

  return {
    type: FETCH_PRODS,
    payload: request
  }
}

export function searchProds (term, category) {
  const request = axios.get(`${ROOT_URL}/products/search/${category}?q=${term}`, config)

  return {
    type: FETCH_PRODS,
    payload: request
  }
}

export function createProd (values, callback) {
  const request = axios
    .post(`${ROOT_URL}/products`, values)
    .then(product => {
      callback(product.data)
      return product
    })
  return {
    type: FETCH_PROD,
    payload: request
  }
}

export function fetchProd (id) {
  const request = axios.get(`${ROOT_URL}/products/${id}`)

  return {
    type: FETCH_PROD,
    payload: request
  }
}

export function deleteProd (id) {
  const request = axios
    .delete(`${ROOT_URL}/products/${id}`)

  return {
    type: DELETE_PROD,
    payload: request
  }
}

export function updateProd (id, values) {
  const request = axios
    .post(`${ROOT_URL}/products/${id}`, values)

  return {
    type: FETCH_PROD,
    payload: request
  }
}
