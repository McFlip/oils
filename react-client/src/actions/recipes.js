import axios from 'axios'
import { ROOT_URL, FETCH_RECIPE, DELETE_RECIPE } from '../constants'

export function fetchRecipe (id) {
  const request = axios.get(`${ROOT_URL}/recipes/${id}`)
  return {
    type: FETCH_RECIPE,
    payload: request
  }
}

export function searchRecipes (term) {
  const request = axios.get(`${ROOT_URL}/recipes/search?q=${term}`)
  return request
}

export function createRecipe (recipe, callback) {
  axios
    .post(`${ROOT_URL}/recipes/create`, recipe)
    .then(r => callback(r.data))
}

export function updateRecipe (id, value) {
  const request = axios.post(`${ROOT_URL}/recipes/${id}/`, value)
  return {
    type: FETCH_RECIPE,
    payload: request
  }
}

export function deleteRecipe (id) {
  const request = axios
    .delete(`${ROOT_URL}/recipes/${id}`)
  return {
    type: DELETE_RECIPE,
    payload: request
  }
}
