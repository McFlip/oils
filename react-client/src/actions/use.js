import axios from 'axios'
import { FETCH_USES, FETCH_USE, SEARCH_USES, DELETE_USE, ADD_USE, ROOT_URL, REMOVE_USE } from '../constants/'

export function fetchUses (id, refType) {
  const request = axios.get(`${ROOT_URL}/uses/${id}/${refType}`)

  return {
    type: FETCH_USES,
    payload: request
  }
}

export function searchUses (term) {
  const request = axios.get(`${ROOT_URL}/uses/search?q=${term}`)

  return {
    type: SEARCH_USES,
    payload: request
  }
}

export function createUse (values) {
  const request = axios.post(`${ROOT_URL}/uses`, values)
  alert(`New Use ${values.title} created`)

  return {
    type: ADD_USE,
    payload: request
  }
}

export function fetchUse (id) {
  const request = axios.get(`${ROOT_URL}/uses/${id}`)
  return request

  /* return {
    type: FETCH_USE,
    payload: request
  }; */
}

export function deleteUse (useId) {
  axios.delete(`${ROOT_URL}/uses/${useId}`)

  return {
    type: DELETE_USE,
    payload: id
  }
}

export function removeUse (id, category, refId) {
  const request = axios.delete(`${ROOT_URL}/uses/${id}/${category}/${refId}`)

  return {
    type: REMOVE_USE,
    payload: request
  }
}

export function addUse (id, category, refId) {
  axios.post(`${ROOT_URL}/uses/${id}/${category}/${refId}`)

  return {
    type: ADD_USE,
    payload: { data: id }
  }
}
