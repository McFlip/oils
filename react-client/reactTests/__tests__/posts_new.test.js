/* eslint-env jest */
import React from 'react'
import { render, newStore, fireEvent } from '../test-utils'
import PostsNew from 'components/posts_new'
import * as postsActionMock from 'actions/posts'

jest.mock('actions/posts')
const store = newStore()

// create post tests
test('render new post', () => {
  const newPost = render(
    <PostsNew
      match={{
        params: { id: 'a' },
        isExact: true,
        path: '',
        url: ''
      }}
      history={{
        location: { state: undefined },
        push: () => null
      }}
    />,
    { store }
  )
  expect(newPost).toMatchSnapshot()
})

test('create new post', () => {
  const { getByText, getByLabelText } = render(
    <PostsNew
      match={{
        params: { id: 'a' },
        isExact: true,
        path: '',
        url: ''
      }}
      history={{
        location: { state: undefined },
        push: () => null
      }}
    />,
    { store }
  )
  // test valid submit
  // relying on HTML5 required prop for this form
  // uploading an image will be covered in end to end tests
  fireEvent.change(getByLabelText('Title For Post'), { target: { value: 'test title' } })
  fireEvent.change(getByLabelText('Post Content'), { target: { value: 'test content' } })
  fireEvent.click(getByText('Submit'))
  expect(postsActionMock.createPost).toHaveBeenCalledTimes(1)
  const formData = postsActionMock.createPost.mock.calls[0][0]
  expect(formData.get('title')).toBe('test title')
  expect(formData.get('content')).toBe('test content')
  expect(formData.get('image')).toBe('null')
})

// edit post tests
test('render edit post', () => {
  const editPost = render(
    <PostsNew
      match={{
        params: { id: 'a', postId: 'postId' },
        isExact: true,
        path: '',
        url: ''
      }}
      history={{
        location: { state: undefined },
        push: () => null
      }}
    />,
    { store }
  )
  const { getByLabelText } = editPost
  expect(postsActionMock.fetchPost).toHaveBeenCalledTimes(1)
  expect(getByLabelText(/title/i).value).toBe('postTitle')
  expect(getByLabelText(/content/i).value).toBe('postContent')
  expect(editPost).toMatchSnapshot()
})

test('edit post', /* async */ () => {
  const store = newStore()
  const { getByText, getByAltText } = render(
    <PostsNew
      match={{
        params: { id: 'a', postId: 'postId' },
        isExact: true,
        path: '',
        url: ''
      }}
      history={{
        location: { state: undefined },
        push: () => null
      }}
    />,
    { store }
  )
  // test valid submit keeping post as is - no changes
  fireEvent.click(getByText('Submit'))
  // createPost was called in previous test
  expect(postsActionMock.createPost).toHaveBeenCalledTimes(1)
  expect(postsActionMock.updatePost).toHaveBeenCalledTimes(1)
  let formData = postsActionMock.updatePost.mock.calls[0][0]
  expect(formData.get('title')).toBe('postTitle')
  expect(formData.get('content')).toBe('postContent')
  expect(formData.get('image')).toBe('null')
  expect(formData.get('deleteImg')).toBe('false')
  // test deleting the image
  fireEvent.click(getByText('Delete'))
  fireEvent.click(getByText('Submit'))
  expect(postsActionMock.updatePost).toHaveBeenCalledTimes(2)
  formData = postsActionMock.updatePost.mock.calls[1][0]
  expect(formData.get('image')).toBe('null')
  expect(formData.get('deleteImg')).toBe('true')
  expect(getByAltText('post image').style._values).toEqual({ 'filter': 'grayscale(100%) blur(5px)' })
  // delete post
  fireEvent.click(getByText('Delete Post'))
  expect(postsActionMock.deletePost).toHaveBeenCalledTimes(1)
  expect(postsActionMock.deletePost.mock.calls[0][0]).toBe('a')
  expect(postsActionMock.deletePost.mock.calls[0][1]).toBe('postId')
})
