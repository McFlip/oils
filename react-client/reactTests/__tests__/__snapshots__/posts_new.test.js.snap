// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render edit post 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="navbar navbar-expand-sm bg-dark navbar-dark fixed-top"
        >
          <a
            class="navbar-brand"
            href="/"
          >
            <img
              alt="Home"
              src="/favicon.ico"
            />
          </a>
          <button
            class="navbar-toggler"
            data-target="#navbarNavDropdown"
            data-toggle="collapse"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="collapse navbar-collapse"
            id="navbarNavDropdown"
          >
            <ul
              class="navbar-nav"
            >
              <li
                class="nav-item active"
              >
                <a
                  class="nav-link"
                  href="/products"
                >
                  products
                </a>
              </li>
              <li
                class="nav-item "
              >
                <a
                  class="nav-link"
                  href="/uses"
                >
                  uses
                </a>
              </li>
              <li
                class="nav-item "
              >
                <a
                  class="nav-link"
                  href="/recipes"
                >
                  recipes
                </a>
              </li>
            </ul>
          </div>
        </nav>
        <form
          enctype="multipart/form-data"
        >
          <div
            class="form-group "
          >
            <label
              for="title"
            >
              Title For Post
            </label>
            <input
              class="form-control"
              id="title"
              name="title"
              required=""
              type="text"
              value="postTitle"
            />
            <div
              class="text-help"
            />
          </div>
          <div
            class="card"
          >
            <h3>
              Existing Image
            </h3>
            <p>
              click delete to remove the image or select a new image to replace it
            </p>
            <button
              class="btn btn-danger"
            >
              Delete
            </button>
            <img
              alt="post image"
              class="post-edit-img"
              src="http://localhost:3000/images/postImage"
            />
          </div>
          <div>
            <label>
              Image
            </label>
            <div>
              <input
                accept=".jpg, .png, .jpeg"
                type="file"
              />
            </div>
          </div>
          <div
            class="form-group "
          >
            <label
              for="content"
            >
              Post Content
            </label>
            <textarea
              class="form-control"
              id="content"
              name="content"
            >
              postContent
            </textarea>
            <div
              class="text-help"
            />
          </div>
          <button
            class="btn btn-primary"
            type="submit"
          >
            Submit
          </button>
          <a
            class="btn btn-secondary"
            href="/products/a"
          >
            Cancel
          </a>
          <button
            class="btn btn-danger float-right"
          >
            Delete Post
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="navbar navbar-expand-sm bg-dark navbar-dark fixed-top"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          <img
            alt="Home"
            src="/favicon.ico"
          />
        </a>
        <button
          class="navbar-toggler"
          data-target="#navbarNavDropdown"
          data-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse"
          id="navbarNavDropdown"
        >
          <ul
            class="navbar-nav"
          >
            <li
              class="nav-item active"
            >
              <a
                class="nav-link"
                href="/products"
              >
                products
              </a>
            </li>
            <li
              class="nav-item "
            >
              <a
                class="nav-link"
                href="/uses"
              >
                uses
              </a>
            </li>
            <li
              class="nav-item "
            >
              <a
                class="nav-link"
                href="/recipes"
              >
                recipes
              </a>
            </li>
          </ul>
        </div>
      </nav>
      <form
        enctype="multipart/form-data"
      >
        <div
          class="form-group "
        >
          <label
            for="title"
          >
            Title For Post
          </label>
          <input
            class="form-control"
            id="title"
            name="title"
            required=""
            type="text"
            value="postTitle"
          />
          <div
            class="text-help"
          />
        </div>
        <div
          class="card"
        >
          <h3>
            Existing Image
          </h3>
          <p>
            click delete to remove the image or select a new image to replace it
          </p>
          <button
            class="btn btn-danger"
          >
            Delete
          </button>
          <img
            alt="post image"
            class="post-edit-img"
            src="http://localhost:3000/images/postImage"
          />
        </div>
        <div>
          <label>
            Image
          </label>
          <div>
            <input
              accept=".jpg, .png, .jpeg"
              type="file"
            />
          </div>
        </div>
        <div
          class="form-group "
        >
          <label
            for="content"
          >
            Post Content
          </label>
          <textarea
            class="form-control"
            id="content"
            name="content"
          >
            postContent
          </textarea>
          <div
            class="text-help"
          />
        </div>
        <button
          class="btn btn-primary"
          type="submit"
        >
          Submit
        </button>
        <a
          class="btn btn-secondary"
          href="/products/a"
        >
          Cancel
        </a>
        <button
          class="btn btn-danger float-right"
        >
          Delete Post
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(observable): [Function],
  },
  "unmount": [Function],
}
`;

exports[`render new post 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="navbar navbar-expand-sm bg-dark navbar-dark fixed-top"
        >
          <a
            class="navbar-brand"
            href="/"
          >
            <img
              alt="Home"
              src="/favicon.ico"
            />
          </a>
          <button
            class="navbar-toggler"
            data-target="#navbarNavDropdown"
            data-toggle="collapse"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="collapse navbar-collapse"
            id="navbarNavDropdown"
          >
            <ul
              class="navbar-nav"
            >
              <li
                class="nav-item active"
              >
                <a
                  class="nav-link"
                  href="/products"
                >
                  products
                </a>
              </li>
              <li
                class="nav-item "
              >
                <a
                  class="nav-link"
                  href="/uses"
                >
                  uses
                </a>
              </li>
              <li
                class="nav-item "
              >
                <a
                  class="nav-link"
                  href="/recipes"
                >
                  recipes
                </a>
              </li>
            </ul>
          </div>
        </nav>
        <form
          enctype="multipart/form-data"
        >
          <div
            class="form-group "
          >
            <label
              for="title"
            >
              Title For Post
            </label>
            <input
              class="form-control"
              id="title"
              name="title"
              required=""
              type="text"
              value=""
            />
            <div
              class="text-help"
            />
          </div>
          <div>
            <label>
              Image
            </label>
            <div>
              <input
                accept=".jpg, .png, .jpeg"
                type="file"
              />
            </div>
          </div>
          <div
            class="form-group "
          >
            <label
              for="content"
            >
              Post Content
            </label>
            <textarea
              class="form-control"
              id="content"
              name="content"
            />
            <div
              class="text-help"
            />
          </div>
          <button
            class="btn btn-primary"
            type="submit"
          >
            Submit
          </button>
          <a
            class="btn btn-secondary"
            href="/products/a"
          >
            Cancel
          </a>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="navbar navbar-expand-sm bg-dark navbar-dark fixed-top"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          <img
            alt="Home"
            src="/favicon.ico"
          />
        </a>
        <button
          class="navbar-toggler"
          data-target="#navbarNavDropdown"
          data-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse"
          id="navbarNavDropdown"
        >
          <ul
            class="navbar-nav"
          >
            <li
              class="nav-item active"
            >
              <a
                class="nav-link"
                href="/products"
              >
                products
              </a>
            </li>
            <li
              class="nav-item "
            >
              <a
                class="nav-link"
                href="/uses"
              >
                uses
              </a>
            </li>
            <li
              class="nav-item "
            >
              <a
                class="nav-link"
                href="/recipes"
              >
                recipes
              </a>
            </li>
          </ul>
        </div>
      </nav>
      <form
        enctype="multipart/form-data"
      >
        <div
          class="form-group "
        >
          <label
            for="title"
          >
            Title For Post
          </label>
          <input
            class="form-control"
            id="title"
            name="title"
            required=""
            type="text"
            value=""
          />
          <div
            class="text-help"
          />
        </div>
        <div>
          <label>
            Image
          </label>
          <div>
            <input
              accept=".jpg, .png, .jpeg"
              type="file"
            />
          </div>
        </div>
        <div
          class="form-group "
        >
          <label
            for="content"
          >
            Post Content
          </label>
          <textarea
            class="form-control"
            id="content"
            name="content"
          />
          <div
            class="text-help"
          />
        </div>
        <button
          class="btn btn-primary"
          type="submit"
        >
          Submit
        </button>
        <a
          class="btn btn-secondary"
          href="/products/a"
        >
          Cancel
        </a>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(observable): [Function],
  },
  "unmount": [Function],
}
`;
